test/ruby/test_pack.rb

```diff
@@ -936,4 +936,90 @@ class Array
       assert_equal "oh no", v
     end;
   end
+
+  def test_pack_unpack_R
+    # ULEB128 encoding (unsigned)
+    assert_equal("\x00", [0].pack("R"))
+    assert_equal("\x01", [1].pack("R"))
+    assert_equal("\x7f", [127].pack("R"))
+    assert_equal("\x80\x01", [128].pack("R"))
+    assert_equal("\xff\x7f", [0x3fff].pack("R"))
+    assert_equal("\x80\x80\x01", [0x4000].pack("R"))
+    assert_equal("\xff\xff\xff\xff\x0f", [0xffffffff].pack("R"))
+    assert_equal("\x80\x80\x80\x80\x10", [0x100000000].pack("R"))
+    assert_equal("\xff\xff\xff\xff\xff\xff\xff\xff\xff\x01", [0xffffffffffffffff].pack("R"))
+
+    # Multiple values
+    assert_equal("\x01\x02", [1, 2].pack("R*"))
+    assert_equal("\x7f\x80\x01", [127, 128].pack("R*"))
+
+    # Negative numbers should raise an error
+    assert_raise(ArgumentError) { [-1].pack("R") }
+    assert_raise(ArgumentError) { [-100].pack("R") }
+
+    # Unpack tests
+    assert_equal([0], "\x00".unpack("R"))
+    assert_equal([1], "\x01".unpack("R"))
+    assert_equal([127], "\x7f".unpack("R"))
+    assert_equal([128], "\x80\x01".unpack("R"))
+    assert_equal([0x3fff], "\xff\x7f".unpack("R"))
+    assert_equal([0x4000], "\x80\x80\x01".unpack("R"))
+    assert_equal([0xffffffff], "\xff\xff\xff\xff\x0f".unpack("R"))
+    assert_equal([0x100000000], "\x80\x80\x80\x80\x10".unpack("R"))
+    assert_equal([0xffffffffffffffff], "\xff\xff\xff\xff\xff\xff\xff\xff\xff\x01".unpack("R"))
```

Comment:
So this is the biggest tested number, but it's only 64 bits right? LEB128 supports arbitrary size, so might make sense to test some much larger numbers?

Also it's perhaps better to do it in ruby-spec directly, but that's another topic.

Reply:
> Would it be that large? What's the ratio of increased compared to that same integer as a Ruby Bignum?

Not sure I follow. I think as a bignum it's just going to be the total number of bytes (maybe?). I just feel weird about allowing an algorithm that continues unbounded (I mean, bounded by the input size, but can possibly read the whole thing).